AmazonPublisherServicesAdapter/src/main/java/com/chartboost/mediation/amazonpublisherservicesadapter/AmazonPublisherServicesAdapter.kt

```diff
@@ -864,3 +775,140 @@ class AmazonPublisherServicesAdapter : PartnerAdapter {
+) : DTBAdInterstitialListener {
+    private var fullscreenAd: DTBAdInterstitial? = null
+
+    fun setAd(ad: DTBAdInterstitial) {
+        fullscreenAd = ad
+    }
+
+    override fun onAdLoaded(adView: View?) {
+        PartnerLogController.log(LOAD_SUCCEEDED)
+        continuationRef.get()?.let {
+            if (it.isActive) {
+                it.resume(
+                    Result.success(
+                        PartnerAd(
+                            ad = fullscreenAd,
+                            details = emptyMap(),
+                            request = request,
+                        ),
+                    ),
+                )
+            }
+        } ?: run {
+            PartnerLogController.log(LOAD_FAILED, "Unable to resume continuation in onAdLoaded. Continuation is null.")
+        }
+    }
+
+    override fun onAdFailed(adView: View?) {
+        PartnerLogController.log(LOAD_FAILED)
+        continuationRef.get()?.let {
+            if (it.isActive) {
+                it.resume(
+                    Result.failure(
+                        ChartboostMediationAdException(ChartboostMediationError.CM_LOAD_FAILURE_UNKNOWN),
+                    ),
+                )
+            }
+        } ?: run {
+            PartnerLogController.log(LOAD_FAILED, "Unable to resume continuation in onAdFailed. Continuation is null.")
+        }
+    }
+
+    override fun onAdClicked(adView: View?) {
+        CoroutineScope(Main).launch {
+            PartnerLogController.log(DID_CLICK)
+            listener?.onPartnerAdClicked(
+                PartnerAd(
+                    ad = fullscreenAd,
+                    details = emptyMap(),
+                    request = request,
+                ),
+            ) ?: run {
+                PartnerLogController.log(
+                    DID_CLICK,
+                    "Unable to fire onPartnerAdClicked for Amazon Publisher Services adapter. Listener is null.",
+                )
+            }
+        }
+    }
+
+    override fun onAdLeftApplication(adView: View?) {
+        // NO-OP
+    }
+
+    override fun onAdOpen(adView: View?) {
+    }
+
+    override fun onAdClosed(adView: View?) {
+        CoroutineScope(Main).launch {
+            PartnerLogController.log(DID_DISMISS)
+            listener?.onPartnerAdDismissed(
+                PartnerAd(
+                    ad = fullscreenAd,
+                    details = emptyMap(),
+                    request = request,
+                ),
+                null,
+            ) ?: run {
+                PartnerLogController.log(
+                    DID_DISMISS,
+                    "Unable to fire onPartnerAdDismissed for Amazon Publisher Services adapter. Listener is null.",
+                )
+            }
+        }
+    }
+
+    override fun onImpressionFired(adView: View?) {
+        CoroutineScope(Main).launch {
+            PartnerLogController.log(DID_TRACK_IMPRESSION)
+            onShowSuccess()
```

Comment:
oh as safety, should we clear the lambda after calling this?

Reply:
I tested that that but saw no differences with anything 🤔